src/server/game/Entities/Creature/Creature.h

```diff
@@ -659,7 +659,7 @@ class Creature : public Unit, public GridObject<Creature>, public MapObject
 
         // Handling caster facing during spellcast
         void SetTarget(ObjectGuid guid) override;
-        void FocusTarget(Spell const* focusSpell, WorldObject const* target);
+        void FocusTarget(Spell const* focusSpell, ObjectGuid targetGuid);
```

Comment:
If caster == target FocusTarget will be called but with no target now. So _focusSpell will have a value. Will fix that asap, still one question: Should I call FocusTarget when the spell does not have a target ? If I do, when the spell is finished ReleaseFocus will be called and creature will start attack the tank again, If I don't.. ehm he won't I guess, so I will need to look at that :p